account_analytic_distribution_manual/models/account_analytic_distribution_manual.py

```diff
@@ -22,6 +32,29 @@ class AccountAnalyticDistributionManual(models.Model):
         ),
     ]
 
+    @api.depends("analytic_distribution")
+    def _compute_analytic_distribution_import(self):
+        aa_model = self.env["account.analytic.account"]
+        for item in self:
+            data = {}
+            distribution = item.analytic_distribution
+            for key in list(distribution.keys()):
+                aa_record = aa_model.browse(int(key))
+                data[aa_record.name] = distribution[key]
+            item.analytic_distribution_import = data
+
+    def _inverse_analytic_distribution_import(self):
+        """Convert the json to the appropriate value of analytic_distribution."""
+        aa_model = self.env["account.analytic.account"]
+        for item in self:
+            data = {}
+            new_distribution = item.analytic_distribution_import
+            for key in list(new_distribution.keys()):
+                aa_record = aa_model.search([("name", "=", key)], limit=1)
```

Comment:
```suggestion
                aa_record = aa_model.search([("name", "=", key), ("company_id", "=", self.env.company.id)], limit=1)
```

I think it is better to add `company_id` to the domain to avoid selecting accounts from other companies (or at least minimize the problem). I created two records with the same name in different companies. The manual distribution belongs to San Francisco. I selected both companies, imported the file, and it picked the account from the other company. Then, I selected only one company and got an error.

![image](https://github.com/user-attachments/assets/7b123be3-53ed-4410-9216-14f4b8d7ba11)

![image](https://github.com/user-attachments/assets/fe923694-bed6-4994-8926-ca79482073f2)


![image](https://github.com/user-attachments/assets/a59c5487-b75a-4853-a243-a134cdaf2123)

Reply:
Thanks, changes done